src/fs/file_info.go

```diff
@@ -25,6 +26,14 @@ type fileInfo struct {
 	mimeType string
 	mtime    time.Time
 	size     int64
+	cache    fileCache
+}
+
+type fileCache struct {
+	status bool
+	mtime  time.Time
+	size   int64
+	data   []byte
```

Comment:
Sending the byte data of 100 x 100 px image inside the files list API will be a bad idea. On an average a thumbnail image will be of 20-30 KB. Assuming a directory contains 100 images (which is reasonable), it will amount to approx 2.5 MB of data in the API response which would take a lot of time to load.
Instead better option would be to make another API for fetching the thumbnail of a given image, and include its url here in the files list API only if the thumbnail exists.